api/v1alpha1/hostedcluster_types.go

```diff
@@ -405,6 +413,22 @@ type ClusterVersionStatus struct {
 	ObservedGeneration int64 `json:"observedGeneration"`
 }
 
+// ClusterConfiguration contains a single global configuration
+// resource for a HostedCluster.
+type ClusterConfiguration struct {
+	// Kind is the API kind of the configuration resource
+	// contained in Content
+	// +kubebuilder:validation:Required
+	// +required
+	Kind string `json:"kind"`
+
+	// Content embeds the configuration resource
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +kubebuilder:validation:Required
```

Comment:
Why do we need `kind`?

I assume  we don't want to `kubebuilder:validation:EmbeddedResource` here since there's no guarantees the management cluster has the apiVersion required by a given content?

Reply:
Yes